### DISCUSSION TITLE
Selectively render card elements based on host value (Javascript)

### DISCUSSION THREAD

**Discussion by etsek** [microsoft/AdaptiveCards#8777] [General]
Is it possible (in Node/Javascript) to determine the card host application at runtime (i.e. Teams desktop vs Teams mobile) so as to display/hide certain parts accordingly? For instance, have an image component only display when the card is rendered in the desktop version of MS Teams.

**Comment by jwoo-msft** [comment]
@etsek, AC wouldn't know of its host. 

Alternatively, you can use an AC templated card, that can be expanded differently based on the custom host argument in the template data. The custom host argument will be set by your code after discovering about the host.

**Reply by etsek** [reply]
Thank you, as a matter of fact I am using AC templating but not sure how the host can be discovered. Are you referring to the $host binding context? Haven't been able to find an example online.

**Reply by jwoo-msft** [reply]
No, I was referring to hosts you were targeting to. You will have to communicate to the receiver of your template to find about the hosts.
